app/services/ZuoraService.scala

```diff
@@ -17,10 +17,33 @@ import scala.concurrent.ExecutionContext.Implicits.global
 import scala.concurrent.Future
 import scala.concurrent.duration._
 
+sealed trait ZuoraFilter {
+  def toFilterString: String
+}
+
+case class SimpleFilter(key: String, value: String) extends ZuoraFilter {
+  override def toFilterString = s"$key='$value'"
+}
+
+case class AndFilter(clauses: (String, String)*) extends ZuoraFilter {
+  override def toFilterString =
+    clauses.map(p => SimpleFilter.tupled(p).toFilterString).mkString(" AND ")
+}
+
+case class OrFilter(clauses: (String, String)*) extends ZuoraFilter {
```

Comment:
It's best to use the same interface for `AndFilter` and `OrFilter`, and in the first case we splat. I find it a bit nicer, and I assume the general case to be splats. But if you prefer I can change that to `Seq`